lighthouse-core/gather/drivers/driver.js

```diff
@@ -328,10 +341,11 @@ class Driver {
     });
   }
 
-  beginNetworkCollect() {
+  beginNetworkCollect(opts) {
     return new Promise((resolve, reject) => {
       this._networkRecords = [];
       this._networkRecorder = new NetworkRecorder(this._networkRecords);
+      this.updateUrlIfRedirected(opts);
```

Comment:
total nit: maybe something like `enableUrlUpdateIfRedirected` to make it clear it's just turning on the listener?